packages/inquirer/src/types.mts

```diff
@@ -1,21 +1,20 @@
 /* eslint-disable @typescript-eslint/no-explicit-any */
 import type {
   input,
-  select,
-  number,
-  confirm,
-  rawlist,
-  expand,
-  checkbox,
-  password,
-  editor,
+  InputConfig,
+  SelectConfig,
+  CheckboxConfig,
+  ConfirmConfig,
+  NumberConfig,
+  RawlistConfig,
+  ExpandConfig,
+  PasswordConfig,
+  EditorConfig,
 } from '@inquirer/prompts';
-import type { Prettify, KeyUnion, DistributiveMerge, Pick } from '@inquirer/type';
-import { Observable } from 'rxjs';
+import type { Prettify } from '@inquirer/type';
+import type { Merge, ValueOf } from 'type-fest';
```

Comment:
Let's inline those in `@inquirer/type` and skip an extra external dependency for types.